fxa/oauth.py

```diff
@@ -222,6 +223,20 @@ def authorize_token(self, sessionOrAssertion, scope=None, client_id=None):
 
         return resp['access_token']
 
+    def _decode_key(self, key, token):
+        pubkey = jwt.algorithms.RSAAlgorithm.from_jwk(key)
+        # The FxA OAuth ecosystem currently doesn't make good use of aud, and instead relies on scope for restricting which services can accept which tokens. So there's no value in checking it here, and in fact if we check it here, it fails because the right audience isn't being requested.
+        decoded = jwt.decode(
+            token, pubkey, algorithms=['RS256'], options={'verify_aud': False}
+        )
```

Comment:
Oh, I almost forgot - per [[1]](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/docs/oauth/jwt-access-tokens.md#local-verification-of-a-jwt-access-token) we must check that the JWT has a `typ` property in the header with value `at+JWT`.

Unfortunately PyJWT doesn't seem to offer a convenient API for doing this check, since the `typ` field is in the header rather than the payload. The best I could come up with was:

```
if jwt.get_unverified_header(token).get('typ') != 'at+JWT':
    raise TrustError
```

(The "unverified" in the name there is slightly unsettling, but it's OK - the header is actually covered by the signature and thus worth checking, it's just that you can't verify the signature without having first read the header!)

Reply:
The document linked says it should be 'at+jwt' lowercase, so I'm going with that.